src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/finalclass/Example1.java

```diff
@@ -6,43 +6,53 @@
 </module>
 */
 
-// xdoc section -- start
-final class Example1 {
-  private Example1() { }
-}
+package com.puppycrawl.tools.checkstyle.checks.design.finalclass;
```

Comment:
@stoyanK7 did we miss template update here?

Reply:
It seems so..

Reply:
Can we create a regexp module to place violations on `<source>` tags in templates, or something similar to catch these? Please make an issue on this.

Reply:
```
checkstyle git:(fix-finalclass-example) ✗ for t in $(find . -name "*.xml.template"); do grep "<source>" $t > /dev/null && echo $t; done | wc -l
59
```

There are probably some valid usages, but this is a start.

Reply:
Why finalclass appeared in this PR ?
There supposed to be only superclone.

Reply:
Rabbit hole of problems, I need to create some issues and break PR up now. PR is still on me, no need to review now.

Reply:
Ci is green , let's merge, and move on and update in separate PRs

Reply:
Ok, let's merge this and move along.